c2rust-transpile/src/translator/operators.rs

```diff
@@ -1034,9 +1034,9 @@ impl<'c> Translation<'c> {
             }
         }?;
 
-        // Unused unary operators (`-foo()`) may have side effects, so we need
-        // to add them to stmts.
-        if ctx.is_unused() {
+        // Some unused unary operators (`-foo()`) may have side effects, so we need
+        // to add them to stmts when arg expr is not pure
```

Comment:
```suggestion
        // to add them to stmts when `arg` is not pure.
```

Reply:
new code will write unused expression when operator is not increment/decrement. so the comment is outdated now.